src/zh/mh1234/src/eu/kanade/tachiyomi/extension/zh/mh1234/MH1234.kt

```diff
@@ -0,0 +1,139 @@
+
+    override fun latestUpdatesFromElement(element: Element) = popularMangaFromElement(element)
+
+    override fun latestUpdatesNextPageSelector() = null
+
+    // Search Page
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList) = GET(
+        baseUrl.toHttpUrl().newBuilder()
+            .addPathSegment("search/")
+            .addQueryParameter("keywords", query)
+            .addQueryParameter("page", page.toString()).build(),
+    )
+
+    override fun searchMangaParse(response: Response): MangasPage {
+        val doc = response.asJsoup()
+        val cur = response.request.url.queryParameter("page")?.toInt() ?: 1
+        val total = doc.selectFirst("#total-page")!!.attr("value").toInt()
+        val mangas = doc.select(popularMangaSelector()).map(::popularMangaFromElement)
+        return MangasPage(mangas, cur < total)
+    }
+
+    override fun searchMangaSelector() = throw UnsupportedOperationException()
+
+    override fun searchMangaFromElement(element: Element) = throw UnsupportedOperationException()
+
+    override fun searchMangaNextPageSelector() = throw UnsupportedOperationException()
+
+    // Manga Detail Page
+
+    override fun mangaDetailsParse(document: Document) = SManga.create().apply {
+        val txtItme = document.select(".txtItme")
+        author = txtItme[0].text()
+        status = when (txtItme[2].selectFirst("a:last-child")?.text()) {
+            "连载中" -> SManga.ONGOING
+            "已完结" -> SManga.COMPLETED
+            else -> SManga.UNKNOWN
+        }
+        val desc = document.selectFirst("#full-des") ?: document.selectFirst("#simple-des")
+        description = desc?.text()?.substring(3)
+    }
+
+    // Manga Detail Page / Chapters Page (Separate)
+
+    // override fun chapterListRequest(manga: SManga) = GET(baseUrl + manga.url)
+
+    override fun chapterListSelector() = throw UnsupportedOperationException()
+
+    override fun chapterFromElement(element: Element) = throw UnsupportedOperationException()
+
+    override fun chapterListParse(response: Response): List<SChapter> {
+        val doc = response.asJsoup()
+        val date = DATE_FORMAT.parse(doc.selectFirst(".txtItme .date")!!.text())?.time
+        val list = doc.select("#chapter-list-1 a").map {
+            SChapter.create().apply {
+                this.setUrlWithoutDomain(it.absUrl("href"))
+                name = it.text()
+                date_upload = date ?: Date().time
+                chapter_number = NUM_REGEX.find(it.text())?.value?.toFloat() ?: -1F
+            }
+        }
+        return list.reversed()
+    }
+
+    // Manga View Page
+
+    override fun pageListParse(document: Document): List<Page> {
+        val body = document.body().html()
+        val prefix = "https://gmh1234.wszwhg.net/${IMG_REGEX2.find(body)!!.groups[1]!!.value}"
+        val list = Json.decodeFromString<List<String>>(IMG_REGEX1.find(body)!!.groups[1]!!.value)
```

Comment:
```suggestion
        val list = IMG_REGEX1.find(body)!!.groups[1]!!.value.parseAs<List<String>>()
```
this uses app provided json instance. (need to import)